### PR TITLE
[ADD] rma_sale_mrp: allow returning phantom products

### PR DESCRIPTION
**Pull request by chienandalu**
When a sale line has a phantom product (mrp kits) the RMA would not be
possible as the wizard couldn't pair the components moves with the
product in the line. With this approach, we can at least return the
spare components of the original kit line.

cc @Tecnativa TT26753

### CODE REVIEW

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Review by ernestotejeda — APPROVED** [review]

**Comment by marcelsavegnago** [comment]
@chienandalu I believe you saw that you have a travis error

File: rma_sale/wizard/sale_order_rma_wizard.py [hunk 1]
```diff
@@ -143,7 +146,7 @@ def _compute_move_id(self):
         for record in self:
             if record.picking_id:
                 record.move_id = record.picking_id.move_lines.filtered(
-                    lambda r: (r.sale_line_id.product_id == record.product_id
+                    lambda r: (r.sale_line_id == record.sale_line_id and r.product_id == record.product_id
```
Comment (pedrobaeza) [inline rma_sale/wizard/sale_order_rma_wizard.py]: This is not correct if there's no linked `sale_line_id` at all.

Comment (chienandalu) [reply rma_sale/wizard/sale_order_rma_wizard.py]: Is there an example in wich that could happen?

Comment (pedrobaeza) [reply rma_sale/wizard/sale_order_rma_wizard.py]: Returning a return will keep the sale_line_id?

Comment (chienandalu) [reply rma_sale/wizard/sale_order_rma_wizard.py]: But this wizard comes from de `sale.order`, how could we be returning from a return?

Comment (pedrobaeza) [reply rma_sale/wizard/sale_order_rma_wizard.py]: We do an RMA, return back to the customer, and a second RMA is needed inside the warranty period.

Comment (chienandalu) [reply rma_sale/wizard/sale_order_rma_wizard.py]: Ok, I understand now. I'll check

Comment (chienandalu) [reply rma_sale/wizard/sale_order_rma_wizard.py]: Ok, just tested it and indeed this approach fails at that point :/

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "restructuredtext.confPath": ""
+}
```
Comment (victoralmau) [inline .vscode/settings.json]:
```suggestion

```

File: rma/views/rma_views.xml [hunk 3]
```diff
@@ -66,6 +66,7 @@
                 <field name="user_id"/>
                 <field name="partner_id"/>
                 <field name="product_id"/>
+
```
Comment (victoralmau) [inline rma/views/rma_views.xml]:
```suggestion
```

File: rma_sale_mrp/tests/test_rma_sale.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    # def test_create_rma_with_so(self):
+    #     rma_form = Form(self.env['rma'])
+    #     rma_form.partner_id = self.partner
+    #     rma_form.order_id = self.sale_order
+    #     rma_form.product_id = self.product_kit
+    #     rma_form.product_uom_qty = 5
+    #     rma = rma_form.save()
+    #     rma.action_confirm()
+    #     self.assertTrue(rma.reception_move_id)
+    #     self.assertFalse(rma.reception_move_id.origin_returned_move_id)
```
Comment (victoralmau) [inline rma_sale_mrp/tests/test_rma_sale.py]: You would need to uncomment this or remove it, don't you think?

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
@ernestotejeda @pedrobaeza Can you review, please?

**Comment by chienandalu** [comment]
and @victoralmau :sweat_smile:

**Review by ernestotejeda — COMMENTED** [review]
Code review :+1:

**Review by victoralmau — APPROVED** [review]

**Comment by victoralmau** [comment]
Travis is red in test

**Comment by chienandalu** [comment]
Yes @victoralmau the test wasn't deterministic enough. I also put some order in the changes to separate the base ones from the new module ones

**Comment by chienandalu** [comment]
This has been in production in customer for a while now. I think we can merge

**Pull request merged**